substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadSleepEvent.java

```diff
@@ -35,8 +36,17 @@
 import com.oracle.svm.core.jfr.JfrNativeEventWriterDataAccess;
 import com.oracle.svm.core.jfr.JfrTicks;
 import com.oracle.svm.core.jfr.SubstrateJVM;
+import jdk.jfr.Category;
+import jdk.jfr.Label;
+import jdk.jfr.Name;
+import jdk.jfr.Timespan;
 
-public class ThreadSleepEvent {
+@Category("Java Application")
+@Label("Java Thread Sleep")
+@Name("jdk.ThreadSleep")
```

Comment:
Does that really work reliably? If I understand everything correctly, then the following is happening:
- HotSpot uses `jdk.jfr.events.ThreadSleepEvent` as the mirror event.
- HotSpot registers `jdk.internal.event.ThreadSleepEvent` as the thread sleep event. This creates a `PlatformEventType` object at build time that ends up in the image heap (instance A).
- You create a new `PlatformEventType` object for the SVM-internal class `com.oracle.svm.core.jfr.events.ThreadSleepEvent` at build time (instance B). This new object reuses the event id of instance A but otherwise behaves like a VM-internal event (e.g., `SubstrateJVM.eventSettings` is used to determine if the event is enabled or disabled).
- You unconditionally enable the thread sleep event at build time (`SubstrateJVM.handleMirrorEvents()`).

I see the following issues:
- What if the event should not be enabled by default?
- What if the event is enabled at runtime via `jdk.internal.event.ThreadSleepEvent`? I don't think that the updated value will be used because JDK events directly store in their `PlatformEventType` object if the event is enabled. However, we would access `SubstrateJVM.eventSettings` to determine if the event is enabled.

Reply:
Thank you for the review Christian! I was under the impression that built in events were enabled by default. I have removed that unconditional enablement and instead substituted `PlatformEventType.setEnabled(boolean)`. This way changes can be propagated to the `SubstrateJVM.eventSettings` if the event happens to be one of the supported mirror event types. Please let me know if this is an acceptable solution.